app.js

```diff
@@ -121,9 +123,20 @@ app.post('/api/registerCurrentData', async (req, res) => {
 })
 
 app.get('/api/getData', async (req, res) => {
-  const fileToSend = await getHighestPriorityFile()
+  let fileToSend
+  // the query data determines which priority we have: newest or oldest first.
+  const priority = req.query.data
+  if (priority === 'old') {
+    console.log('1')
+    fileToSend = await getHighestPriorityFile(true)
+  } else if (priority === 'new') {
+    console.log('2')
+    fileToSend = await getHighestPriorityFile(false)
+  } else {
+    console.log('3')
+    fileToSend = -1
+  }
   if (fileToSend === -1) {
-    res.status(400)
```

Comment:
Ich glaube hier müsste trotzdem ein 200 Status Code geworfen werden, da die Anfrage richtig war. Wenn es kein Element dazu gibt, ist Respone einfach leer. 400 sollte geworfen werden, wenn in der Query etwas steht, was nicht old oder new ist. Dann wurde eine "Bad request" gestellt.